### DISCUSSION TITLE
Creating Deno Stack

### DISCUSSION THREAD

**Discussion by CordialGit** [denoland/deno#14466] [Ideas]
There are these **buzzwords** that always catches developers' attention. 

And subconsciously, these buzzwords have also contributed to pulling people to certain Technologies-- because people interpreted those buzzwords to mean that the underlying Tech has matured enough, to be able to pull stable integrations.

Thus we have these Tech Stacks known as:
--MERN Stack
-- MEAN Stack
-- MEVN Stack
-- LAMP Stack
-- Django Stack
-- Java Stack.

Focusing on the JavaScript Stacks, MERN Stack suggests MongoDB, Express, React and Node.Js.

The MEAN Stack is Angular, and the MEVN Stack is Vue.JS.

I am proposing to the Deno Team, that we need to have a DENO Stack. This Stack should be called a **DORM** Stack.

This DORM Stack is Deno, Oak, React, and MongoDB.

Regards

**Comment by sunw31** [comment]
I am really recommend to copy nestjs-stack.

Deno-Stack
Some candidated framework, obey the fast-software-wins rule...
https://alosaur.com/
reason: https://github.com/denosaurs/bench

Deno-Admin
Easy to use cms template, like rails-admin、django-admin、jhispster-admin...

Deno-Anaconda http://scipy-lectures.org/
Whole data life-cycle, from request to database to data analysis...
Let developers just describe their  schema and entity-relationship, fast to create a application...

Deno-Microservice
Deno-Guide like python-guide.org nodejs.dev
Deno-Landscape like CNCF landscape https://landscape.cncf.io/
StateofDeno like stateofjs https://2021.stateofjs.com/en-US/features

Deno Web3 Stack
Ethersjs、Foundry-rs

Deno DAO
Deno NFT
Deno Token
Deno Swap

**Reply by CordialGit** [reply]
If I say I understand everything you said, I lie. It's like you got me totally lost.

Perhaps you didn't get my question well, or maybe I didn't ask my question well.

The question will now be-- Can Deno, Oak, React and MongoDB work smoothly together?

And can **Production-Ready** Apps be built with this combo of Deno, Oak, React and MongoDB?

Regards

**Reply by sunw31** [reply]
oh, you mean standard deno stack like ruby on rails, php on laravel , java on spring , nodejs on nestjs , elixir on phoenix, golang on goframe, 
right?

**Reply by Sorikairox** [reply]
Another alternative to Alosaur, it's called Danet, based on Oak and is closer to NestJs.

It is not officially production ready, but it will be soon. We're missing:
- lifecycle hooks, ETA this Sunday. 
- HTML template engine's management, ETA next 2 weeks.
- an awesome high level documentation
- a starter project repo

But I'm biased, I'm the author of Danet. If anyone wanna join this journey to build a savory web framework.

**Reply by CordialGit** [reply]
> oh, you mean standard deno stack like ruby on rails, php on laravel , java on spring , nodejs on nestjs , elixir on phoenix, golang on goframe, right?

Yes. But Node.JS always works with Express.JS and MongoDB.

I am speaking of a standard Deno Stack along the lines of Deno-Oak-React-MongoDB.

**Reply by Sorikairox** [reply]
What do you mean by `NodeJS always works with Express.JS and MongoDB` ?

The MEAN stack is purely a community invention. Neither Express nor Mongo has official approval or support from Node to be a standard.
A lot of people are using Koa, Fastify, and SQL engines on Node. 

If you want to create a stack, feel free, to write articles about it and promote it, and the community may adopt and promote it (I certainly will, I like the idea)

**Reply by sunw31** [reply]
deno can borrow nodejs experiences. 

nodejs modern web stack:
nestjs、prisma、nextjs on vercel

image deno modern web stack:
Danet or Alosaur or Oak、 xxxORM,、fresh on Deno Deploy

github.com/Sorikairox/Danet
github.com/lucacasonato/fresh

so orm is the big problem

**Comment by sunw31** [comment]
![image](https://user-images.githubusercontent.com/13845339/174442295-1bd5c9dd-4d73-4d9e-af35-0e0a2fcf5c79.png)


https://github.com/deepkit/deepkit-framework
